MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -8682,4 +8689,90 @@ public IUnitRating getUnitRating() {
 		
 		return unitRating;
 	}
+
+    public int getPeacetimeCost() {
+        int cost = 0;
+
+        cost += getPayRoll(getCampaignOptions().useInfantryDontCount());
+        cost += getMonthlySpareParts();
+        cost += getMonthlyFuel();
+        cost += getMonthlyAmmo();
+        return cost;
+    }
+
+    public long getMonthlySpareParts() {
+        long partsCost = 0;
+
+        for (Unit u : getUnits()) {
+            if (u.isMothballed()) {
+                continue;
+            }
+            partsCost += u.getSparePartsCost();
+        }
+        return partsCost;
+    }
+
+    public long getMonthlyFuel() {
+        long fuelCost = 0;
+
+        for (Unit u : getUnits()) {
+            if (u.isMothballed()) {
+                continue;
+            }
+            fuelCost += u.getFuelCost();
+        }
+        return fuelCost;
+    }
+
+    public long getMonthlyAmmo() {
+        long ammoCost= 0;
+
+        for (Unit u : getUnits()) {
+            if (u.isMothballed()) {
+                continue;
+            }
+            ammoCost += u.getAmmoCost();
+        }
+        return ammoCost;
+    }
+
+    public boolean isFactionComstar() {
+        if (getFactionCode().equals("CS")) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean isFactionClan() {
+        switch (getFactionCode()) {
```

Comment:
This has the potential to throw a NPE.  You should do a null check before the switch statement.

Reply:
Same, though honestly if faction returned null then there is an even bigger bug somewhere else. the faction code should never return null in this case. See line 340 of this file.

Reply:
While the faction code is not supposed to return a null, it's possible that future code could change this.  In any case, it's a best practice to do a null-check before invoking a switch statement.  Better to detect and gracefully handle the null along with logging a sensible error than allowing the switch statement to blow up with a generic NPE.

Reply:
I decided to re-use the Faction.java mechanism for identifying whether a faction is clan, to save on duplicated logic. The campaign's current faction is always set to something by default (based on the campaign's constructor), so I don't think a null check is really worthwhile.

Reply:
Unfortunately, setFactionCode doesn't contain null protection, so it is possible to set the factionCode null.  Technically, the better solution would, indeed, be to make sure that factionCode is only ever set through setFactionCode and add a null check there.